hoomd/md/ActiveForceComputeGPU.cu

```diff
@@ -57,57 +56,21 @@ __global__ void gpu_compute_active_force_set_forces_kernel(const unsigned int gr
 
     unsigned int tag = d_groupTags[group_idx];
     unsigned int idx = d_rtag[tag];
+    unsigned int type = __scalar_as_int(d_pos[idx].w);
 
-    Scalar3 f;
-    Scalar3 t;
-    // rotate force according to particle orientation only if orientation is linked to active force vector
-    if (orientationLink == true)
-        {
-        vec3<Scalar> fi;
-        f = make_scalar3(d_f_actMag[tag] * d_f_actVec[tag].x,
-                        d_f_actMag[tag] * d_f_actVec[tag].y, d_f_actMag[tag] * d_f_actVec[tag].z);
-        quat<Scalar> quati(d_orientation[idx]);
-        fi = rotate(quati, vec3<Scalar>(f));
-        d_force[idx].x = fi.x;
-        d_force[idx].y = fi.y;
-        d_force[idx].z = fi.z;
-
-        vec3<Scalar> ti;
-        t = make_scalar3(d_t_actMag[tag] * d_t_actVec[tag].x,
-                        d_t_actMag[tag] * d_t_actVec[tag].y, d_t_actMag[tag] * d_t_actVec[tag].z);
-        ti = rotate(quati, vec3<Scalar>(t));
-        d_torque[idx].x = ti.x;
-        d_torque[idx].y = ti.y;
-        d_torque[idx].z = ti.z;
+    Scalar3 f = make_scalar3(d_f_actMag[type]*d_f_actVec[type].x, d_f_actMag[type]*d_f_actVec[type].y, d_f_actMag[type]*d_f_actVec[type].z);
```

Comment:
I don't expect this kernel to take very long to run as is, so this micro optimization isn't necessary. If you want to refactor the storage, we could reduce the number of memory transactions by packing the vector and magnitude into a single Scalar4 and reading it with ``__ldg``:
```
Scalar4 act = __ldg(d_f_act + type);
vec3<Scalar> f(act.x * act.w, act.y * act.w, act.z * act.w)
```